astropy/coordinates/tests/test_solar_system.py

```diff
@@ -397,10 +399,11 @@ def test_ephemeris_wrong_input(ephemeris, expected_error):
 
 @pytest.mark.skipif(not HAS_JPLEPHEM, reason="requires jplephem")
 def test_ephemeris_local_file_not_ephemeris():
-    # NOTE: This test currently leaves the file open (ResourceWarning).
-    # To fix this issue, an upstream fix is required in jplephem
-    # package.
-    with pytest.warns(ResourceWarning), pytest.raises(ValueError, match="^file starts"):
+    if minversion("jplephem", "2.23"):
+        ctx = nullcontext()
+    else:
+        ctx = pytest.warns(ResourceWarning)
```

Comment:
I'd prefer to keep a short comment to clarify that the `ResourceWarning` is because of an upstream bug. And perhaps it would be better to use `pytest.mark.filterwarnings` instead of `pytest.warns`.